src/sentry/testutils/cases.py

```diff
@@ -384,11 +384,10 @@ class RuleTestCase(TestCase):
     def get_event(self):
         return self.event
 
-    def get_rule(self, data=None):
-        return self.rule_cls(
-            project=self.project,
-            data=data or {},
-        )
+    def get_rule(self, **kwargs):
```

Comment:
`get_rule` doesn't return `Rules`. It returns things that extend `RuleBase` like `Action` or `Condition` types (which are also not rules.) I considered renaming a lot of these things for clarity but I don't think it's worth it as part of this commit (see above comment on refactoring.)

I updated the rest of the tests to explicitly pass `data=` here, as well as `rule=` when necessary.